### ISSUE TITLE
REST Support for na_ontap_firmware_upgrade

### ISSUE THREAD

**Issue by mamoep** [ansible-collections/netapp.ontap#36]
### Summary

na_ontap_firmware_upgrade needs console permissions to work. We rely on domain credentials for personalized access, which cannot have console permissions. 
REST endpoint for software & firmware download is available. Please add REST support to the module so console permissions are not necessary.

### Component Name

na_ontap_firmware_upgrade

**Comment by lonico** [comment]
I'm not aware of a REST API to download the firmware (there is one to download the software).

But we could remove the limitation on console permissions by using the REST CLI passthrough as an alternative to the ZAPI system-cli.

**Comment by mamoep** [comment]
The API reference for /cluster/software/download mentions both software and firmware. I quick test with 9.7 shows that it is working with disk firmware.

Example playbook:
```yaml
- hosts: localhost
  become: false
  gather_facts: false
  collections:
    - netapp.ontap
  name: Test firmware download
  vars:
    username: admin
    password: foo
    login: &login
      hostname: "clustername"
      username: "{{ username }}"
      password: "{{ password }}"
      https: true
      validate_certs: false

  tasks:
  - name: test disk firmware update via native REST call
    na_ontap_restit:
      api: cluster/software/download
      method: POST
      body:
        url: "https://server.local/all.zip"
      <<: *login
    connection: local
    register: result

  - name: get job
    na_ontap_restit:
      api: "cluster/jobs/{{ result.response.job.uuid }}"
      method: GET
      <<: *login
    connection: local
    register: job

  - debug:
      var: job
```

Output:
```
TASK [test disk firmware update via native REST call] *****************************************************************************************************************************
changed: [localhost]

TASK [get job] ********************************************************************************************************************************************************************
changed: [localhost]

TASK [debug] **********************************************************************************************************************************************************************
ok: [localhost] => {
    "job": {
        "changed": true,
        "failed": false,
        "response": {
            "code": 10551382,
            "description": "POST /api/cluster/software/download",
            "message": "Package download is still running.",
            "start_time": "2021-11-23T09:15:07+01:00",
            "state": "running",
            "uuid": "7832d6c0-4c35-11ec-91f0-00a0989b61f5"
        },
        "status_code": 200
    }
}
```

**Comment by lonico** [comment]
My bad.  We tend to use `software` in a very restrictive way.

**Comment by lonico** [comment]
fixed by aef51a931ebfdd5b593f4e9ca8871cac17b404e0
will be released as 21.15.0
